### ISSUE TITLE
Rename RCUK and GSRT projects

### ISSUE THREAD

**Issue by marekhorst** [openaire/iis#1250]
We should integrate this pull request:
https://github.com/openaire/iis/pull/1242

renaming:
- RCUK->UKRI
- GSRT->GSRI

This does not have to go through the data curation process because the change is rather trivial. We should just align our integration tests to reflect new projects naming.

**Comment by przemyslawjacewicz** [comment]
@johnfouf I'm integrating your PR #1242 into IIS integration test suite and I've noticed something I wanted to clarify. We have integration test cases for both 'old' funder names, that is RCUK and GSRT. After merging your PR both test cases failed. I thought that the change should not break current mining. That is, in the case of RCUK -> UKRI change we would mine projects related to RCUK (as before) and UKRI (new ones), possible assigning the same funder name for both. This is not the case now, because we will not be able to mine projects with RCUK funding class - we consider only UKRI funding class in `projects.sql`
```sql
--iis-wf/iis-wf-referenceextraction/src/main/resources/eu/dnetlib/iis/wf/referenceextraction/project/main_sqlite/oozie_app/lib/scripts/projects.sql:39

select jdict('documentId', docid, 'projectId', id, 'confidenceLevel', sqroot(min(1.49,confidence)/1.5), 'textsnippet', (prev||" <<< "||middle||" >>> "||next)) as C1, docid, id, fundingclass1, grantid from
( select prev,middle,next,docid,id,max(confidence) as confidence, docid, id,  fundingclass1, grantid from ( select 
(0.3
+hardmatch*0.6
+subrcukmatch
+rcukmatch)
as confidence, docid, id,prev,middle,next, fundingclass1, grantid
from (
unindexed select regexprmatches('\bRCUK\b|\bUKRI\b|[Rr]esearch [Cc]ouncils UK', context) as rcukmatch,
regexprmatches('\b'||rcuk_subfunder||'\b', context) as subrcukmatch,
regexprmatches('(?:G\d{6,7})|(?:[A-Z]{2}\/\w{6,7}\/\d{1,2}(?:\/xx)?)|(?:(?:BBS|PPA)\/[A-Z](?:\/[A-Z])?\/(?:\w{8,9}|(?:\d{4}\/)?\d{5}(?:\/\d)?))|(?:(?:RES|PTA)-\d{3}-\d{2}-\d{4}(?:-[A-Z]|-\d{1,2})?)|(?:MC_(?:\w{8,10}|\w{2}_(?:\w{2,4}_)?(?:\d{4,5}|[UG]\d{7,9})(?:\/\d{1,2})?))|(?:MC_\w{2}_\w{2}(?:_\w{2})?\/\w{7}(?:\/\d)?)|(?:ESPA-[A-Z]{3,6}-\d{4}(?:-[A-Z]{3}-\d{3}|-\d{3})?)', middle) as hardmatch,
docid, id, fundingclass1, grantid,middle,prev,next
from (
select docid, stripchars(middle,'.)(,[]') as middle, prev, next, prev||' '||middle||' '||next as context
from (
  setschema 'docid,prev,middle,next' select c1, textwindow2s(c2,15,1,1,'(?:[SG]?\d{5,7}(?:\/\d)?)|(?:[A-Z]{2}\/\w{6,7}\/\d{1,2}(?:\/xx)?)|(?:[GE]\d{2,3}\/\d{1,4})|(?:(?:BBS|PPA)\/[A-Z](?:\/[A-Z])?\/(?:\w{8,9}|(?:\d{4}\/)?\d{5}(?:\/\d)?))|(?:(?:RES|PTA)-\d{3}-\d{2}-\d{4}(?:-[A-Z]|-\d{1,2})?)|(?:MC_(?:\w{8,10}|\w{2}_(?:\w{2,4}_)?(?:\d{4,5}|[UG]\d{7,9})(?:\/\d{1,2})?))|(?:MC_\w{2}_\w{2}(?:_\w{2})?\/\w{7}(?:\/\d)?)|(?:[A-Za-z]{3,9}\d{5,7}a?)|(?:ESPA-[A-Z]{3,6}-\d{4}(?:-[A-Z]{3}-\d{3}|-\d{3})?)') from pubs where c2 is not null
)), grants
WHERE fundingclass1="UKRI" and middle = grantid --ONLY UKRI funding class
) where confidence>0.3 ) group by docid,id)
```
I've checked input project datastore from latest PROD run for RCUK and UKRI funding classes and out of 2849155 total projects we have 115670 RCUK projects and only 137 UKRI ones. The same consideration applies for GSRT -> GSRI change. Is this the proper way we should handle this mining ?

**Comment by johnfouf** [comment]
Oh! @przemyslawjacewicz as I understand it does not crash but it is a matter of a number of 115670 RCUK projects which fail to be mined right? As far as I know, the funder's name has changed so we expected that both old and new projects would have the new name in both cases. Probably, we have to clarify it with Miriam.

**Comment by johnfouf** [comment]
@przemyslawjacewicz one more question, perhaps projects list has not been yet updated in PROD? There are also still RCUK projects in beta input?

**Comment by przemyslawjacewicz** [comment]
> Oh! @przemyslawjacewicz as I understand it does not crash but it is a matter of a number of 115670 RCUK projects which fail to be mined right? As far as I know, the funder's name has changed so we expected that both old and new projects would have the new name in both cases. Probably, we have to clarify it with Miriam.

Yes, the changes in `projects.sql` do not crash the run, that is mining runs without any 'runtime' problems but RCUK and GSRT projects are not found anymore. Only UKRI and GSRI projects are found. 

> @przemyslawjacewicz one more question, perhaps projects list has not been yet updated in PROD? There are also still RCUK projects in beta input?

Yes, there are still 'old' projects in BETA, of total 3014283 projects we have 113934 RCUK and 110257 UKRI together with 3243 GSRT and 0 GSRI projects.

**Comment by harry-di** [comment]
Thanks @przemyslawjacewicz but the numbers of input projects does not add up, so I'm asking a few questions to Miriam on Slack and when she answers we'll have a clearer picture.
In BETA, we assumed that she duplicated projects (so I would expect number to be the same, i.e. RCUK=UKRI and GRST=GSRI), in for the mining updating to be tested on the new project names but the old mining could still... Once the updated mining was OK, then she would delete the old projects.
In other words, "RCUK and GSRT projects are not found anymore" is correct mining behaviour.
She is replying now...

**Comment by harry-di** [comment]
Here is Miriam's input:

"_For the number for RCUK / UKRI. We should not have ukri in production. I have not added the api there and (as far as I am concerned) I did not modify the 8980 in production so it should not depend on direct indexing and they should not be there. In beta there should be both projects with nsp UKRI and nsp RCUK. For the numbers in beta, when building the production graph we consider both infrastructures, so the one for production is also loaded and the RCUK projects come from there. In beta there are no GSRI projects because they have been blacklisted and so not imported from the aggregator while loading the data.
...
I can remove the grsi namespace prefix from the black list. In this way it will be visible also
I though I should wait with gsri, but I must have misunderstood_"

So our understanding is that the UKRI projects and the RCUK ones are duplicated; they come from the same api
the only thing that changes is the nsp. So we can ignore RCUK projects during mining and get only UKRI ones.
Similarly with GRSI (ignore GSRT) but let's wait after Miriam removes the blacklisting of GRSI projects.

All the above concerns BETA. That's were we need to test first. Ignore PROD for now...plus the 137 UKRI in PROD is a mistake that Miriam is trying to figure out (it is shown in Explore as a funding steam).

**Comment by przemyslawjacewicz** [comment]
@harry-di Thanks for very exact explanation!! I'll remove RCUK and GSRT test cases and replace them with UKRI and GSRI cases. The changes will be merged into master before the release of IIS so the new version will be ready for mining using new names.

**Comment by harry-di** [comment]
@przemyslawjacewicz By the way, Miriam just reported:
Harry I have started today the provision run for production
if the loading of the data from the aggregator is needed for beta, it has to wait

**Comment by przemyslawjacewicz** [comment]
@johnfouf I'm having problems with GSRT/GSRI mining. Since there are no GSRI projects and I couldn't find any GSRI mentions in document texts (in BETA) I artificially changed the funding class of existing test case project from `GSRT::` to `GSRI::`. But the project is not found. The same change in RCUK/UKRI case results in success - the project is found with 'new' funder name. I think this means that either I'm missing something or the GSRT/GSRI part of the changes does not work. These are test case data:
* project
```json
{
  "id": "40|gsri________::574ef85ac608988be5167363bb257010",
  "projectGrantId": "447985",
  "projectAcronym": null,
  "fundingClass": "GSRI::",
  "jsonextrainfo": "{}"
}
```
* document text
```json
{
  "text": "This work was supported by grants from the General Secretariat of Research and Technology of Greece (GSRT) SINERGASIA “THERACAN” 11SYN-1-485 and KRIPIS “STHENOS” MIS 447985.",
  "id": "50|dedup_wf_001::92685f7e22342478a67376985a967618"
}
```
* document to project result
```json
{
   "documentId":"50|dedup_wf_001::92685f7e22342478a67376985a967618",
   "projectId":"40|gsri________::574ef85ac608988be5167363bb257010",
   "confidenceLevel":0.8164966,
   "textsnippet":" work supported grants general secretariat research technology greece _gsrt_ sinergasia _theracan_ 11syn_1_485 kripis _sthenos_ <<< 447985 >>>   "
}
```
Could you verify this case ? Thanks.

**Comment by johnfouf** [comment]
Oh no! there was also another place with "and fundingclass1='GSRT')" which didn't changed!

**Comment by johnfouf** [comment]
Thanks

**Comment by przemyslawjacewicz** [comment]
Thanks, I'll check the patched version.

**Comment by przemyslawjacewicz** [comment]
Tests passed, patch was successful! Integration with master branch is on the way.
